apps/api/src/bid/controllers/bid/bid.controller.ts

```diff
@@ -0,0 +1,36 @@
+import { BidHttpService } from "@akashnetwork/http-sdk";
+import assert from "http-assert";
+import flatten from "lodash/flatten";
+import { singleton } from "tsyringe";
+
+import { AuthService, Protected } from "@src/auth/services/auth.service";
+import { ListBidsResponse } from "@src/bid/http-schemas/bid.schema";
+import { UserWalletRepository } from "@src/billing/repositories";
+
+@singleton()
+export class BidController {
+  constructor(
+    private readonly bidHttpService: BidHttpService,
+    private readonly authService: AuthService,
+    private readonly userWalletRepository: UserWalletRepository,
+  ) {}
+
+  @Protected([{ action: "create", subject: "DeploymentSetting" }])
```

Comment:
why `DeploymentSetting`?

```suggestion
  @Protected([{ action: "sign", subject: "UserWallet" }])
```

Reply:
based on this https://github.com/akash-network/console/blob/main/apps/api/src/billing/controllers/wallet/wallet.controller.ts#L40

Reply:
No particular reason, It was just me picking an arbitrary permission. Fixed it.